pkg/action/release_testing.go

```diff
@@ -110,17 +112,38 @@ func (r *ReleaseTesting) GetPodLogs(out io.Writer, rel *release.Release) error {
 	for _, h := range rel.Hooks {
 		for _, e := range h.Events {
 			if e == release.HookTest {
-				req := client.CoreV1().Pods(r.Namespace).GetLogs(h.Name, &v1.PodLogOptions{})
-				logReader, err := req.Stream(context.Background())
-				if err != nil {
-					return errors.Wrapf(err, "unable to get pod logs for %s", h.Name)
+				var (
+					podList *corev1.PodList
+					err     error
+				)
+				switch h.Kind {
+				case "Job":
+					podList, err = client.CoreV1().Pods(r.Namespace).List(context.Background(), metav1.ListOptions{
+						LabelSelector: labels.FormatLabels(map[string]string{"job-name": h.Name}),
+					})
+				case "Pod":
+					podList, err = client.CoreV1().Pods(r.Namespace).List(context.Background(), metav1.ListOptions{
+						FieldSelector: labels.FormatLabels(map[string]string{"metadata.name": h.Name}),
+					})
 				}
 
-				fmt.Fprintf(out, "POD LOGS: %s\n", h.Name)
-				_, err = io.Copy(out, logReader)
-				fmt.Fprintln(out)
 				if err != nil {
-					return errors.Wrapf(err, "unable to write pod logs for %s", h.Name)
+					return errors.Wrapf(err, "unable to get pod list for %s/%s", h.Kind, h.Name)
+				}
+
+				for _, pod := range podList.Items {
+					req := client.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{})
```

Comment:
I think this issue existed previously, but it seems like there's no way to configure which container to get logs for? (It'll always be the first container in the pod I think?)

Reply:
It can be configured, but if write the name of the first container, it will have the same effect as before.